File: packages/javascript/lib/verify/isEmail.ts
export function isEmail(email: string, onlyLowerCase = false): boolean {
	// RFC2822 Email Validation
	const char = `a-z${onlyLowerCase ? '' : 'A-Z'}`;

	const regex = new RegExp(
		`^[${char}0-9!#$%&'*+/=?^_\`{|}~-]+(?:\\.[${char}0-9!#$%&'*+/=?^_\`{|}~-]+)*@(?:[${char}0-9](?:[${char}0-9-]*[${char}0-9])?\\.)+[${char}0-9](?:[${char}0-9-]*[${char}0-9])?$`,
		'g'
	);

	return regex.test(email);
}

File: packages/python/qsu/verify/isEmail.py
import re


def isEmail(email: str, onlyLowerCase: bool = False) -> bool:
	# RFC2822 Email Validation
	char = 'a-z' if onlyLowerCase else 'a-zA-Z'

	pattern = (
		r"^[" + char + r"0-9!#$%&'*+/=?^_`{|}~-]+"
		r"(?:\.[" + char + r"0-9!#$%&'*+/=?^_`{|}~-]+)*"
		r"@(?:[" + char + r"0-9](?:[" + char + r"0-9-]*[" + char + r"0-9])?\.)+"
		r"[" + char + r"0-9](?:[" + char + r"0-9-]*[" + char + r"0-9])?$"
	)

	return re.search(pattern, email) is not None
